### PR TITLE
ST 2/Patch - NaN quantity and uni label color

### PR DESCRIPTION
**Pull request by Tbrid**

### CODE REVIEW

File: packages/webapp/src/components/Task/TaskComplete/HarvestComplete/HarvestUses.js [hunk 1]
```diff
@@ -108,7 +113,7 @@ export default function PureHarvestUses({
 
       <Label style={{ marginBottom: '16px' }}>
         {t('TASK.AMOUNT_TO_ALLOCATE')}
-        <UnitLabel style={{ marginLeft: '12px' }} amount={amount_to_allocate} unitLabel={unit} />
+        <UnitLabel style={{ marginLeft: '12px' }} amount={amount_to_allocate} unitLabel={display_unit[system]} />
```
Comment (beetai) [inline packages/webapp/src/components/Task/TaskComplete/HarvestComplete/HarvestUses.js:116]: Why is this change necessary? It seems to me that `unit` is still being passed in as a prop with the value `persistedFormData?.actual_quantity_unit?.label`. Is that not being used?

**Review by beetai — COMMENTED** [review]
looks good to me, basically ready to approve. just got a comment.

File: packages/webapp/src/components/Task/TaskComplete/HarvestComplete/Quantity.js [hunk 2]
```diff
@@ -41,6 +42,12 @@ export default function PureHarvestCompleteQuantity({
   const ACTUAL_HARVEST_QUANTITY_UNIT = 'actual_quantity_unit';
 
   const disabled = !isValid;
+
+  useEffect(() => {
+    if (getValues(ACTUAL_HARVEST_QUANTITY) === undefined) {
+      setValue(ACTUAL_HARVEST_QUANTITY, task.harvest_task.projected_quantity);
+    }
+  }, []);
```
Comment (beetai) [inline packages/webapp/src/components/Task/TaskComplete/HarvestComplete/Quantity.js:50]: Do you reckon that this could be set in the `useForm`'s defaultValue parameter?

**Review by beetai — COMMENTED** [review]
Sorry, got another comment.

**Review by JimmyRowland — APPROVED** [review]
